File: php/features.php
<?php
declare(strict_types=1);

// KimiQuotes SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class KimiQuotesFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new KimiQuotesBaseFeature();
            case "test":
                return new KimiQuotesTestFeature();
            default:
                return new KimiQuotesBaseFeature();
        }
    }
}

File: py/features.py
# KimiQuotes SDK feature factory

from feature.base_feature import KimiQuotesBaseFeature
from feature.test_feature import KimiQuotesTestFeature


def _make_feature(name):
    features = {
        "base": lambda: KimiQuotesBaseFeature(),
        "test": lambda: KimiQuotesTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
